src/components/Images.vue

```diff
@@ -271,57 +279,158 @@ export default {
+    },
+    async deleteImage(obj) {
+      const options = {
+        message:   `Delete image ${ obj.imageName }:${ obj.tag }?`,
+        type:      'question',
+        buttons:   ['Yes', 'No'],
+        defaultId: 1,
+        title:     'Confirming image deletion',
+        cancelId:  1
+      };
+      const result = await ipcRenderer.invoke('show-message-box', options);
 
-      if (this.$refs.fullWindow) {
-        this.$nextTick(() => {
-          this.$refs.fullWindow.parentElement.parentElement.scrollTop = this.$refs.fullWindow.scrollHeight;
-        });
+      if (result.response === 1) {
+        return;
       }
-    },
-    deleteImage(obj) {
       this.currentCommand = `delete ${ obj.imageName }:${ obj.tag }`;
       this.mainWindowScroll = this.$refs.fullWindow.parentElement.parentElement.scrollTop;
       this.startRunningCommand('delete');
-      ipcRenderer.send('confirm-do-image-deletion', obj.imageName.trim(), obj.imageID.trim());
+      ipcRenderer.send('do-image-deletion', obj.imageName.trim(), obj.imageID.trim());
     },
     doPush(obj) {
       this.currentCommand = `push ${ obj.imageName }:${ obj.tag }`;
+      this.mainWindowScroll = this.$refs.fullWindow.parentElement.parentElement.scrollTop;
       this.startRunningCommand('push');
       ipcRenderer.send('do-image-push', obj.imageName.trim(), obj.imageID.trim(), obj.tag.trim());
     },
     doBuildAnImage() {
-      this.currentCommand = `build ${ this.imageToBuild }`;
+      const imageName = this.imageToBuild.trim();
+
+      this.currentCommand = `build ${ imageName }`;
       this.fieldToClear = 'imageToBuild';
+      this.postCloseOutputWindowHandler = () => this.scrollToImageOnSuccess(imageName);
       this.startRunningCommand('build');
-      ipcRenderer.send('do-image-build', this.imageToBuild.trim());
+      ipcRenderer.send('do-image-build', imageName);
     },
     doPullAnImage() {
-      this.currentCommand = `pull ${ this.imageToPull }`;
+      const imageName = this.imageToPull.trim();
+
+      this.currentCommand = `pull ${ imageName }`;
       this.fieldToClear = 'imageToPull';
+      this.postCloseOutputWindowHandler = () => this.scrollToImageOnSuccess(imageName);
       this.startRunningCommand('pull');
-      ipcRenderer.send('do-image-pull', this.imageToPull.trim());
+      ipcRenderer.send('do-image-pull', imageName);
+    },
+    /**
+     * syntax of a fully qualified tag could start with <hostname>:<port>/
+     * so a colon precedes a tag only if its followed only by valid tag characters
+     * @param fullImageName {string}
+     * @returns {[string, string]}
+     */
+    parseFullImageName(fullImageName) {
+      const m = /^(.+?):([-._A-Za-z0-9]+)$/.exec(fullImageName);
+
+      return m ? [m[1], m[2]] : [fullImageName, 'latest'];
+    },
+    getImageByNameAndTag(imageName, tag) {
+      return this.images.find(image => image.imageName === imageName &&
+        (image.tag === tag || (image.tag === '<none>' && tag === 'latest')));
```

Comment:
Not sure that `<none>` test is correct here; I could do a tagless build then do a pull to get the real `latest`…

Of course I suspect the correct fix is to make sure we don't make such silly image names :)

Reply:
So `parseFullImageName` returns a tag of `latest` if none is specified.

If you actually specify `foo:latest` then the `image.tag === tag` part of the expression will be true, and the second part is ignored.

If you don't specify a tag, it will show up as '<none>' in the images table, but the virtual tag will be `latest`, and the second part of the expression succeeds.